snapcraft/tests/test_sources.py

```diff
@@ -310,6 +310,15 @@ def test_pull_when_target_is_file(self):
         self.mock_symlink.assert_called_once_with(
             '/home/ubuntu/sources/snap/source', 'source_file')
 
+class MockOptions:
+
+    def __init__(self, source=None, source_type=None, source_branch=None,
+                 source_tag=None, source_subdir=None):
+        self.source = source
+        self.source_type = source_type
+        self.source_branch = source_branch
+        self.source_tag = source_tag
+        self.source_subdir = source_subdir
```

Comment:
Instead of duplicating the code, it would be nice if you move the one from test_base_plugin to snapcraft/tests.py and use it like tests.MockOptions.